src/rate_limiter.rs

```diff
@@ -167,3 +168,54 @@ impl Default for RateLimiter {
         Self::new()
     }
 }
+
+#[cfg(feature = "use-mock-crust")]
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use messages::Request;
+    use rand;
+    use types::MessageId;
+
+    fn sleep_millis(time: u64) {
+        ::fake_clock::FakeClock::advance_time(time);
+    }
+
+    fn generate_get_request_payload() -> Vec<u8> {
+        unwrap!(serialisation::serialise(&UserMessage::Request(Request::GetIData {
+                                                                   name: rand::random(),
+                                                                   msg_id: MessageId::new(),
+                                                               })))
+    }
+
+    #[test]
+    fn rate_limiter_check() {
+        let mut rate_limiter = RateLimiter::new();
+        let client_1 = IpAddr::from([0, 0, 0, 0]);
+        let get_req_payload = generate_get_request_payload();
+        let fill_full_iterations = CAPACITY / CLIENT_GET_CHARGE;
+        for _ in 0..fill_full_iterations {
+            assert!(rate_limiter
+                        .add_message(1, client_1, 1, 0, &get_req_payload)
+                        .is_ok());
+        }
+
+        let client_2 = IpAddr::from([1, 1, 1, 1]);
+        assert!(rate_limiter
+                    .add_message(1, client_2, 1, 0, &get_req_payload)
+                    .is_err());
+        sleep_millis(100);
+        assert!(rate_limiter
+                    .add_message(10, client_2, 1, 0, &get_req_payload)
+                    .is_ok());
+
+        sleep_millis(400);
+        // The failure is because all zero payload will be parsed as a `Refresh` message.
```

Comment:
I think this will no longer be the case once https://github.com/maidsafe/maidsafe_utilities/pull/108 is published.  It would need to be all zeros but exactly the right length too.